internal/controller/rabbitmq/rabbitmquser_controller.go

```diff
@@ -0,0 +1,418 @@
+		// Individual permission fields (Configure/Write/Read) are guaranteed to have values
+		// either from user input or from kubebuilder defaults (".*" for full permissions).
+		err = apiClient.SetPermissions(vhostName, username,
+			instance.Spec.Permissions.Configure,
+			instance.Spec.Permissions.Write,
+			instance.Spec.Permissions.Read)
+		if err != nil {
+			instance.Status.Conditions.Set(condition.FalseCondition(rabbitmqv1.RabbitMQUserReadyCondition, condition.ErrorReason, condition.SeverityWarning, rabbitmqv1.RabbitMQUserReadyErrorMessage, err.Error()))
+			return ctrl.Result{}, err
+		}
+	}
+
+	instance.Status.SecretName = secretName
+	instance.Status.Username = username
+	instance.Status.Vhost = vhostName
+	instance.Status.VhostRef = instance.Spec.VhostRef // Track the vhost CR name for finalizer management
+	instance.Status.Conditions.MarkTrue(rabbitmqv1.RabbitMQUserReadyCondition, rabbitmqv1.RabbitMQUserReadyMessage)
+	instance.Status.Conditions.MarkTrue(condition.ReadyCondition, condition.ReadyMessage)
+
+	return ctrl.Result{}, nil
+}
+
+func (r *RabbitMQUserReconciler) reconcileDelete(ctx context.Context, instance *rabbitmqv1.RabbitMQUser, h *helper.Helper) (ctrl.Result, error) {
+	Log := log.FromContext(ctx)
+
+	// If TransportURL finalizer exists, wait for TransportURL to remove it
+	// The TransportURL controller manages this finalizer and removes it when switching users
+	if controllerutil.ContainsFinalizer(instance, rabbitmqv1.TransportURLFinalizer) {
+		return ctrl.Result{RequeueAfter: time.Duration(2) * time.Second}, nil
+	}
+
+	// Remove per-user finalizer from vhost if it exists
+	// Use VhostRef from status (current) or spec (fallback) to find the vhost
+	vhostRef := instance.Status.VhostRef
+	if vhostRef == "" {
+		vhostRef = instance.Spec.VhostRef
+	}
+	if vhostRef != "" {
+		userFinalizer := rabbitmqv1.UserVhostFinalizerPrefix + instance.Name
+		vhost := &rabbitmqv1.RabbitMQVhost{}
+		if err := r.Get(ctx, types.NamespacedName{Name: vhostRef, Namespace: instance.Namespace}, vhost); err == nil {
+			if controllerutil.ContainsFinalizer(vhost, userFinalizer) {
+				controllerutil.RemoveFinalizer(vhost, userFinalizer)
+				if err := r.Update(ctx, vhost); err != nil {
+					// Requeue to retry - we want to clean up the vhost finalizer
+					Log.Error(err, "Failed to remove finalizer from vhost during user deletion, requeueing", "vhost", vhostRef, "finalizer", userFinalizer)
+					return ctrl.Result{RequeueAfter: time.Duration(2) * time.Second}, err
+				}
+			}
+		} else if !k8s_errors.IsNotFound(err) {
+			// Only log non-NotFound errors - if vhost is gone, that's fine
+			Log.Error(err, "Failed to get vhost for finalizer removal during user deletion", "vhost", vhostRef)
+			// Don't block deletion if vhost doesn't exist
+		}
+	}
+
+	username := instance.Status.Username
+	if username == "" {
+		username = instance.Spec.Username
+		if username == "" {
+			username = instance.Name
+		}
```

Comment:
do we need this wir the defaulting in https://github.com/openstack-k8s-operators/infra-operator/pull/511/files#diff-c4cee0a4e6abfb93e34d9f8543be2b84269c0d7805dbda19bcfab131a07ec02eR41 ?